test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1314 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Tests the C/CPP API for obtaining maximum tile size across all fragments of an array.
+ */
+
+#include <test/support/tdb_catch.h>
+#include "test/support/src/helpers.h"
+#include "tiledb/common/stdx_string.h"
+#include "tiledb/sm/buffer/buffer.h"
+#include "tiledb/sm/c_api/tiledb_struct_def.h"
+#include "tiledb/sm/cpp_api/tiledb"
+#include "tiledb/sm/cpp_api/tiledb_experimental"
+#include "tiledb/sm/misc/constants.h"
+#include "tiledb/sm/misc/utils.h"
+#include "tiledb/sm/serialization/array.h"
+
+#include <numeric>
+
+struct MaxTileSizeFx {
+  const std::string main_array_name_ = "max_tile_size_array";
+
+  bool showing_data_ = false;
+
+  MaxTileSizeFx()
+    : vfs_(ctx_) {
+  }
+
+  void remove_temp_dir(const std::string& path) {
+    if (vfs_.is_dir(path.c_str())) {
+      vfs_.remove_dir(path.c_str());
+    }
+  };
+
+  /*
+   * Array creation, Write, routine naming scheme as follows:
+   * 
+   * [(as useful) template <dim/attr types for most expected reuseability flexibility>]
+   * create_[sparse|dense]_[dimensionality]_[series-of-dim-descriptions]_[series-of-attribute-descriptions]
+   * 
+   * The word 'array' was specifically left out as it is reasonably implied from dense/sparse, and 
+   * would seem to simply add additional verbiage to be parsed without providing any additional
+   * enlightenment.
+   * 
+   * [dimesionality] could be omitted, but did seem to provide likely value that otherwise would
+   * require the (mental) parsing of the series-of-dim-descriptions to determine the dimensionality.
+   * 
+   * The series-of-dim-descriptions and series-of-attr-descriptions currently use dN and aN to
+   * (i) identifiy which dimension/attribute is being described and its order of creation
+   * as well as (ii) identifying the specific names of the dimension/attribute being created.  
+   * Those names could be substituted with something else ("row"/"col", "volume") but would 
+   * lose some of the at-a-glance numbered ordering of the dimension/attribute creation.
+   * 
+   * The write routines will be similarly named with at least one 'write' match for each of 
+   * the 'create' routines.
+   * 
+   ***********************************************
+   * 
```

Comment:
Agreed. Please remove the template since it is only used once and only applicable to strings. Rather than templating single cases within a test case, we should work towards higher-level APIs in the test support lib in order to make tests as concise and easy to write as possible.